tests/includes/rest/class-test-inbox-controller.php

```diff
@@ -231,4 +231,262 @@ public function test_get_item() {
 	public function test_get_item_schema() {
 		// Controller does not implement get_item_schema().
 	}
+
+	/**
+	 * Test creating an inbox item with blog user context.
+	 *
+	 * @covers ::create_item
+	 */
+	public function test_create_item_with_blog_user() {
+		\add_filter( 'activitypub_defer_signature_verification', '__return_true' );
+
+		\update_option( 'activitypub_actor_mode', ACTIVITYPUB_BLOG_MODE );
+
+		$blog_actor = \Activitypub\Collection\Actors::get_by_id( \Activitypub\Collection\Actors::BLOG_USER_ID );
+
+		// Set up mock action.
+		$inbox_action = new \MockAction();
+		\add_action( 'activitypub_inbox', array( $inbox_action, 'action' ) );
+
+		$json = array(
+			'id'     => 'https://remote.example/@id',
+			'type'   => 'Create',
+			'actor'  => 'https://remote.example/@test',
+			'object' => array(
+				'id'        => 'https://remote.example/post/test',
+				'type'      => 'Note',
+				'content'   => 'Hello, World!',
+				'to'        => array( $blog_actor->get_id() ),
+				'published' => '2020-01-01T00:00:00Z',
+			),
+			'to'     => array( $blog_actor->get_id() ),
+		);
+
+		$request = new \WP_REST_Request( 'POST', '/' . ACTIVITYPUB_REST_NAMESPACE . '/inbox' );
+		$request->set_header( 'Content-Type', 'application/activity+json' );
+		$request->set_body( \wp_json_encode( $json ) );
+
+		$response = \rest_do_request( $request );
+		$this->assertEquals( 202, $response->get_status() );
+
+		// Verify the action was triggered exactly once for a single recipient.
+		$this->assertEquals( 1, $inbox_action->get_call_count() );
+
+		\remove_filter( 'activitypub_defer_signature_verification', '__return_true' );
+	}
+
+	/**
+	 * Test creating an inbox item with multiple recipients.
+	 *
+	 * @covers ::create_item
+	 */
+	public function test_create_item_with_multiple_recipients() {
+		\add_filter( 'activitypub_defer_signature_verification', '__return_true' );
+
+		\update_option( 'activitypub_actor_mode', ACTIVITYPUB_ACTOR_AND_BLOG_MODE );
+
+		$user_id    = self::factory()->user->create( array( 'role' => 'author' ) );
```

Comment:
Let's delete the users that are being created in this and the following tests. Or create a single user in `set_up_before_class()` (and remove it).